lib/private/Authentication/TwoFactorAuth/Db/ProviderUserAssignmentDao.php

```diff
@@ -91,32 +91,30 @@ public function persist(string $providerId, string $uid, int $enabled): void {
 	/**
 	 * Delete all provider states of a user and return the provider IDs
 	 *
-	 * @param string $uid
-	 *
 	 * @return list<array{provider_id: string, uid: string, enabled: bool}>
 	 */
 	public function deleteByUser(string $uid): array {
 		$qb1 = $this->conn->getQueryBuilder();
 		$selectQuery = $qb1->select('*')
 			->from(self::TABLE_NAME)
 			->where($qb1->expr()->eq('uid', $qb1->createNamedParameter($uid)));
-		$selectResult = $selectQuery->execute();
+		$selectResult = $selectQuery->executeQuery();
 		$rows = $selectResult->fetchAll();
 		$selectResult->closeCursor();
 
 		$qb2 = $this->conn->getQueryBuilder();
 		$deleteQuery = $qb2
 			->delete(self::TABLE_NAME)
 			->where($qb2->expr()->eq('uid', $qb2->createNamedParameter($uid)));
-		$deleteQuery->execute();
+		$deleteQuery->executeStatement();
 
-		return array_map(function (array $row) {
+		return array_values(array_map(function (array $row) {
 			return [
-				'provider_id' => $row['provider_id'],
-				'uid' => $row['uid'],
-				'enabled' => (int) $row['enabled'] === 1,
+				'provider_id' => (string)$row['provider_id'],
+				'uid' => (string)$row['uid'],
+				'enabled' => $row['enabled'] === 1,
```

Comment:
```suggestion
				'enabled' => ((int) $row['enabled']) === 1,
```
? Some DBs only return strings

Reply:
Ahh, that's why oci test are failing. Thanks!